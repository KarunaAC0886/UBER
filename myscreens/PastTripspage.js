import { View, Text, TouchableOpacity } from 'react-native'
import React from 'react'
import styles from './styles'
const PastTripspage = () => {
  return (
    <View style={styles.PastTrippg_main}>
      <View style={styles.PastTrippg_main_contnr}>
      <Text style={styles.PastTrippg_content_frstln}>You havenâ€™t taken a trip yet</Text>
      <View style={styles.PastTrippg_retry_btn_contr}>
      <TouchableOpacity style={styles.PastTrippg_retry_btn}>
      <Text style={styles.PastTrippg_retry_btn_txt}>Retry</Text>
      </TouchableOpacity>
      </View>
      </View>
    </View>
  )
}

export default PastTripspage