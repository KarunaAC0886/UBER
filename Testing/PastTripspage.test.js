import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import PastTripspage from '../myscreens/PastTripspage';

jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: 'FontAwesome5',
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('Past Trips', () => {
  it('Past Trips page renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<PastTripspage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});