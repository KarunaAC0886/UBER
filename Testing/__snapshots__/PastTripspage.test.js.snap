// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Past Trips Past Trips page renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1A1A1A",
      "flex": 0,
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "height": "40%",
        "width": "70%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 20,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      You havenâ€™t taken a trip yet
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 0,
          "flexDirection": "row",
          "height": "40%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#ffffff",
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 18,
            }
          }
        >
          Retry
        </Text>
      </View>
    </View>
  </View>
</View>
`;
